lib/superstore/associations/builder/association.rb

```diff
@@ -1,11 +1,11 @@
 module Superstore::Associations::Builder
   class Association
-    def self.build(model, name, options)
-      new(model, name, options).build
+    def self.build(model, name, **options)
+      new(model, name, **options).build
     end
 
     attr_reader :model, :name, :options
-    def initialize(model, name, options)
+    def initialize(model, name, **options)
```

Comment:
This keeps this call working?

```ruby
Association.new(Model, 'whatever_name', option1: 'abc')
```

Maybe we shouldn't support that and it's better to force `{}` as the last argument in this case.

```ruby
Association.new(Model, 'whatever_name', { option1: 'abc' })
```

I know it's not 'ruby-ish', but it seems like that definition is changing for Ruby 3.0.

Reply:
Yeah I wish we could do something like that second option - passing explicit hashes, or even better moving away from options hashes totally and just using keyword args. The trouble arises when we are interfacing with other methods/libraries that are using the double splat syntax in their method params. 

This write up helped me more than the release notes for Ruby 2.7 - https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/. 

And here are a few examples I used to get my head around it:

```
def run(name, **options)
  puts name, options
end

run('first', { a: 'a' })

# /Users/johnk/Desktop/build.rb:46: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
# /Users/johnk/Desktop/build.rb:42: warning: The called method `run' is defined here

run('second', b: 'b')

# No warnings! Happy Ruby!
```

```
def runza_one(name, options)
  runza_two(name, options)
end

def runza_two(name, scope = nil, **options)
  puts "Name: #{name}"
  puts "Scope: #{scope}"
  puts "Options: #{options}"
end

runza_one('first', a: 'a')

# First time through we get errors, because Ruby doesn't want to force convert our Hash into keyword args when calling the second method
# /Users/johnk/Desktop/build.rb:65: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
# /Users/johnk/Desktop/build.rb:68: warning: The called method `runza_two' is defined here

def runza_one(name, **options)
  runza_two(name, options)
end

runza_one('second', b: 'b')

# Second time through we still have errors. Ruby refuses to force convert that Hash, even though it came in via a double splat operator.
# /Users/johnk/Desktop/build.rb:81: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
# /Users/johnk/Desktop/build.rb:68: warning: The called method `runza_two' is defined here

def runza_one(name, options)
  runza_two(name, **options)
end

runza_one('third', c: 'c')

# All good, no warnings!
```